src/_premake_main.lua

```diff
@@ -333,6 +333,62 @@
 		end
 	end
 
+---
+-- Run non-action generation part.
+---
+	local function find_git_root(current_path)
+		current_path = path.getabsolute(current_path or _MAIN_SCRIPT_DIR)
+	
+		while not os.isdir(path.join(current_path, ".git")) do
+			current_path = path.getdirectory(current_path)
+			if current_path == "" then
+				error("No git root path")
+			end
+		end
+		return current_path
+	end
+
+	local function print_git_post_checkout_hooks(root_path, mode)
```

Comment:
What would be a better place for those functions?

Reply:
Maybe a new file dedicated to git integration?

Reply:
New file created